docs/code-generation/index.md

```diff
@@ -21,17 +21,26 @@ each integration scenario in the best way possible.
 
 Currently, we support:
 
-* [HCL generation](./generate-hcl.md) with stack [context](#generation-context).
-* [File generation](./generate-file.md) with `root` and `stack` [context](#generation-context).
+* [HCL generation](./generate-hcl.md) with `stack` context.
+> This strategy generates code that is closely tied to and relative to the stack where it is defined.
+It provides access to a rich set of code generation features, including globals, metadata, functions, lets,
+and assertions. By using these features you can easily manipulate and transform data within stack context.
+>
+
+* [File generation](./generate-file.md) with `root` and `stack` context.
+ > With this strategy, you can generate code at the project root level, outside of specific stacks. It offers 
+ access to project metadata, functions, lets, and assertions. BY using this relatively broader context you 
+ can generate code that spans multiple stacks or interacts with data at the project level.
+ >
```

Comment:
"With this strategy, you can generate code at the project root level, outside of specific stacks."
This depends on the context, with `context=stack` it actually generates inside stacks (exactly same behavior of `generate_hcl`).
With this approach, it looks like choosing `generate_file` or `generate_hcl` are related to where you'd need the file generated, which is not the case, they are orthogonal concepts.
We have two generate methods (_hcl and _file) and we support 2 execution contexts (`stack` and `root`). It's just a limitation that at the moment the `generate_hcl` does not support `context=root` but this can change in the future.

"... or interacts with data at the project level." 
I don't get what this part refers to.